### PR TITLE
Refactor virtualizer layouts to remove Spectrum-specific logic and improve backward compatibility

### PR DESCRIPTION
**Pull request by devongovett**
This removes Spectrum-specific logic from the layouts in `@react-stately/layout`, and moves that into subclasses within each Spectrum component. These include:

* Loading and empty states
* Overflow overrides
* Height adjustments
* Section header collection differences

It also adds backward compatibility for the `layout` prop of `useTable` which I previously removed. This is done by shimming the `LayoutDelegate` interface using a copied interface that matches the old Virtualizer Layout class so we don't need a dependency on it.

I also reverted the change to use `overflow: clip` because it resulted in incorrect height measurements (caught by Chromatic). This was originally because I thought it introduced additional generic containers in Firefox accessibility tree. I later realized that these only seem to affect VoiceOver which is not a supported combination. Firefox NVDA seems fine.

Finally I removed the workaround added 2 years ago for sticky positioning crashing in Chrome 105 (#3504). I think it's been fixed for long enough that we don't need this anymore.

### CODE REVIEW

File: packages/react-aria-components/test/Table.test.js [hunk 1]
```diff
@@ -837,14 +837,14 @@ describe('Table', () => {
     
     rows = getAllByRole('row');
     expect(rows).toHaveLength(8);
-    expect(rows.map(r => r.textContent)).toEqual(['FooBar', 'Foo 7Bar 7', 'Foo 8Bar 8', 'Foo 9Bar 9', 'Foo 10Bar 10', 'Foo 11Bar 11', 'Foo 12Bar 12', 'Foo 13Bar 13']);
+    expect(rows.map(r => r.textContent)).toEqual(['FooBar', 'Foo 8Bar 8', 'Foo 9Bar 9', 'Foo 10Bar 10', 'Foo 11Bar 11', 'Foo 12Bar 12', 'Foo 13Bar 13', 'Foo 14Bar 14']);
```
Comment (devongovett) [inline packages/react-aria-components/test/Table.test.js:840]: RAC table doesn't add `+ 1` for bottom row borders anymore as Spectrum does.

**Comment by snowystinger** [comment]
Based on your description, i guess https://github.com/adobe/react-spectrum/pull/6611#top is unneeded?

**Review by snowystinger — APPROVED** [review]
Did a quick sweep of ListView and TableView as well as RAC virtualized Table and List, looks good, no crashes or anything dire

**Comment by devongovett** [comment]
Yeah looks like the nubbin isn't clipped anymore on this PR.

**Pull request merged**